### ISSUE TITLE
Autoloader performance issue

### ISSUE THREAD

**Issue by nmoinvaz** [auth0/wp-auth0#461]
It appears that the autoloader slows down the wordpress site significantly.

Running performance monitor I was able to see that for every class that was being searched for, it would search in a bunch of different directories - even for files that were not part of the auth0 plugin. The more plugins I had installed, the worse the problem would get.

![image](https://user-images.githubusercontent.com/1364432/39883893-f93f5fb0-543c-11e8-9834-cfde41d1918c.png)

I believe the issue exists here:

https://github.com/auth0/wp-auth0/blob/9713e6a7cdf4c3446632be5338426dfba102bf3a/WP_Auth0.php#L412

And this is the fix that I did that seems to help things although I don't know if it is a permanent solution:

```
foreach ( $paths as $p ) {
    foreach ( $exts as $ext ) {
        if (strpos($p.$class.$ext, "WP_Auth0") !== FALSE && file_exists( $p.$class.$ext ) ) {
            require_once $p.$class.$ext;
            return true;
        }
    }
}
```

Thanks.

**Comment by joshcanhelp** [comment]
@nmoinvaz - Thank you very much for the thorough report here. I'm going to take a look today and should be able to get this into the next release, coming in the next couple of weeks.

**Comment by joshcanhelp** [comment]
👆 twitched on the close issue button

Alright, I can confirm that this is trying to autoload all classes that are called in plugins. Moving this to high priority to fix and should have a PR ready next week at some point. 

@nmoinvaz thanks again for the heads-up here. Your solution is close but:

- You should check for the name at the very top of that method, exit early. 
- There are a few classes that do not start with `WP_Auth0` ... the JWT verifier library for one (used during login) called `JWT` and the Twitter API library (only used with social sharing, IIRC) called `TwitterAPIExchange`)

**Comment by nmoinvaz** [comment]
I have tested it and it works well so far.

**Comment by joshcanhelp** [comment]
Fixed in #465, merged into `dev` and ready for release 👍
